### PR TITLE
Diaspora: Better handling for unrelayed comments and likes

### PR DESCRIPTION
**Pull request by annando**
This is the next improvement for the Diaspora protocol. We now check more reliable for not relayed comments and likes that have to be send directly to the origin. Additionally retractions for comments and likes (that were meant for Friendica generated threads) will work again.

### CODE REVIEW

File: include/diaspora.php [hunk 1]
```diff
@@ -382,15 +383,19 @@ public static function dispatch_public($msg) {
 	 *
 	 * @return int The message id of the generated message, "true" or "false" if there was an error
 	 */
-	public static function dispatch($importer, $msg) {
+	public static function dispatch($importer, $msg, $fields = null) {
```
Comment (rabuzarus) [inline include/diaspora.php]: please add `* @param object $fields` to the doxygen header

File: include/diaspora.php [hunk 2]
```diff
@@ -456,7 +461,7 @@ public static function dispatch($importer, $msg) {
 	 *
 	 * @return bool Is the posting valid?
 	 */
-	private static function valid_posting($msg, &$fields) {
+	private static function valid_posting($msg) {
```
Comment (rabuzarus) [inline include/diaspora.php]: please remove ` * @param object $fields SimpleXML object that contains the posting when it is valid` from the doxgen header

File: include/diaspora.php [hunk 3]
```diff
@@ -552,7 +557,7 @@ private static function valid_posting($msg, &$fields) {
 
 		// Only some message types have signatures. So we quit here for the other types.
 		if (!in_array($type, array("comment", "message", "like")))
```
Comment (Hypolite) [inline include/diaspora.php]: Standards: Please add braces to this condition.

File: include/diaspora.php [hunk 4]
```diff
@@ -575,7 +584,7 @@ private static function valid_posting($msg, &$fields) {
 			logger("No valid author signature for author ".$fields->author. " in type ".$type." - signed data: ".$signed_data." - Message: ".$msg["message"]." - Signature ".$author_signature, LOGGER_DEBUG);
 			return false;
 		} else
```
Comment (Hypolite) [inline include/diaspora.php]: Standards: Please add braces to this condition.

**Pull request merged**